rpcs3/Emu/SysCalls/lv2/sys_tty.cpp

```diff
@@ -22,7 +22,17 @@ s32 sys_tty_write(s32 ch, vm::ptr<const void> buf, u32 len, vm::ptr<u32> pwritel
 {
 	sys_tty.Log("sys_tty_write(ch=%d, buf_addr=0x%x, len=%d, preadlen_addr=0x%x)", ch, buf.addr(), len, pwritelen.addr());
 
-	if(ch > 15 || (s32)len <= 0) return CELL_EINVAL;
+	if (ch > 15)
+	{
+		sys_tty.Error("sys_tty_write(): specified channel was higher than 15.");
+		return CELL_OK;
+	}
+
+	if (len == 0)
+	{
+		sys_tty.Error("sys_tty_write(): specified length was 0.");
+		return CELL_OK;
+	}
```

Comment:
I don't understand what you didn't understand...
`if (ch > 15) return CELL_EINVAL;` -- This is ok, I mentioned only len check.
`if ((s32)len <= 0) return CELL_OK;` -- This it what actually happens. Even though `len` type is `u32`, it's checked as if it were `s32`.